modopt/tests/test_signal.py

```diff
@@ -1,322 +1,233 @@
-        )
-
-    def tearDown(self):
-        """Unset test parameter values."""
-        self.data1 = None
-        self.data2 = None
-        self.data3 = None
-        self.data4 = None
-        self.data5 = None
-
-    def test_add_noise_poisson(self):
-        """Test add_noise with Poisson noise."""
-        np.random.seed(1)
-        npt.assert_array_equal(
-            noise.add_noise(self.data1, noise_type='poisson'),
-            self.data2,
-            err_msg='Incorrect noise: Poisson',
-        )
-
-        npt.assert_raises(
-            ValueError,
-            noise.add_noise,
-            self.data1,
-            noise_type='bla',
-        )
-
-        npt.assert_raises(ValueError, noise.add_noise, self.data1, (1, 1))
-
-    def test_add_noise_gaussian(self):
-        """Test add_noise with Gaussian noise."""
+        ]
+    )
+    data4 = np.array([[0, 0, 0], [0, 0, 5.0], [6.0, 7.0, 8.0]])
+    data5 = np.array(
+        [[0, 0, 0], [0, 0, 0], [1.0, 2.0, 3.0]],
+    )
+
+    @pytest.mark.parametrize(
+        ("data", "noise_type", "sigma", "data_noise"),
+        [
+            (data1, "poisson", 1, data2),
+            (data1, "gauss", 1, data3),
+            (data1, "gauss", (1, 1, 1), data3),
+            failparam(data1, "fail", 1, data1),
+        ],
+    )
+    def test_add_noise(self, data, noise_type, sigma, data_noise):
+        """Test add_noise."""
         np.random.seed(1)
         npt.assert_almost_equal(
-            noise.add_noise(self.data1),
-            self.data3,
-            err_msg='Incorrect noise: Gaussian',
+            noise.add_noise(data, sigma=sigma, noise_type=noise_type), data_noise
         )
 
-        np.random.seed(1)
-        npt.assert_almost_equal(
-            noise.add_noise(self.data1, sigma=(1, 1, 1)),
-            self.data3,
-            err_msg='Incorrect noise: Gaussian',
-        )
-
-    def test_thresh_hard(self):
-        """Test thresh with hard threshold."""
-        npt.assert_array_equal(
-            noise.thresh(self.data1, 5),
-            self.data4,
-            err_msg='Incorrect threshold: hard',
-        )
-
-        npt.assert_raises(
-            ValueError,
-            noise.thresh,
-            self.data1,
-            5,
-            threshold_type='bla',
-        )
-
-    def test_thresh_soft(self):
-        """Test thresh with soft threshold."""
+    @pytest.mark.parametrize(
+        ("threshold_type", "result"),
+        [("hard", data4), ("soft", data5), failparam("fail", None, raises=ValueError)],
+    )
+    def test_thresh(self, threshold_type, result):
+        """Test threshold."""
         npt.assert_array_equal(
-            noise.thresh(self.data1, 5, threshold_type='soft'),
-            self.data5,
-            err_msg='Incorrect threshold: soft',
+            noise.thresh(self.data1, 5, threshold_type=threshold_type), result
         )
 
-
-class PositivityTestCase(TestCase):
-    """Test case for positivity module."""
```

Comment:
Same question regarding the test case.

Reply:
postiivity module was only one test,  for consistency I added the class encapsulation in c17d3c89d8738b985b1b75f6a8ad8d60a2b0a058